src/lib/dnd-backend.test.js

```diff
@@ -0,0 +1,214 @@
+/* global it, expect */
+import DnDBackend from './dnd-backend'
+
+function makeEntry (item) {
+  if (item.isFile) {
+    // eslint-disable-next-line
+    item.file = (cb) => cb({
+      fullPath: item.fullPath,
+      name: item.name,
+      isDirectory: false,
+      isFile: true
+    })
+  }
+
+  if (item.isDirectory) {
+    item.createReader = () => ({
+      readEntries: function (cb) {
+        if (!this.called) {
+          this.called = true
+          cb(item.childs.map(makeEntry))
```

Comment:
nitpick: use `children`, it's the (special case) pluralisation, and is more familiar as it's often used in react.
```suggestion
          cb(item.children.map(makeEntry))
```